src/cargo/ops/cargo_add/mod.rs

```diff
@@ -934,7 +934,7 @@ fn populate_available_features(
     }
 
     let possibilities = loop {
-        match registry.query_vec(&query, QueryKind::Fuzzy) {
+        match registry.query_vec(&query, QueryKind::Exact) {
```

Comment:
The test in the first commit should be passing on its own and this commit should update the test so its still passing.

By the commit title for the first test, I'm assuming it isn't passing.  The intention of this pattern is
- Tests always pass, making bisection easier
- This shows to you and reviewers that the test does what it intended
- This shows to your reviewers what behavior change this PR causes

Reply:
For this, should that initial test just contain one package within it so that we don't hit this issue?

Reply:
When I do this, I have the test verify we are hitting the failure mode.

For example for #13164
- https://github.com/rust-lang/cargo/pull/13164/commits/7200646a9794468d466a65267d1a33ce872cfed6 adds a lot of tests that pass
- https://github.com/rust-lang/cargo/pull/13164/commits/179135368991f442dc7770bc0a0a0ba044368444 changed the behavior.  In this case, it is showing the error changing from one message to another but in your case it would be an error changing to success.